core/src/main/scala/weaponregex/model/regextree/predefinedCharClassNode.scala

```diff
@@ -12,11 +12,22 @@ import weaponregex.model.Location
 case class PredefinedCharClass(charClass: String, override val location: Location)
     extends Leaf(charClass, location, """\""")
 
-/** POSIX character class leaf node
+/** Unicode character class leaf node
   * @param property
   *   The class character property
+  * @param propValue
+  *   The value of the [[property]]
   * @param location
   *   The [[weaponregex.model.Location]] of the node in the regex string
   */
-case class POSIXCharClass(property: String, override val location: Location, isPositive: Boolean = true)
-    extends Leaf(property, location, if (isPositive) """\p{""" else """\P{""", "}")
+case class UnicodeCharClass(
+    property: String,
+    override val location: Location,
+    isPositive: Boolean = true,
+    propValue: String = ""
```

Comment:
Shouldn't this be an Option? From what I understand an empty string here means the propValue is absent

Reply:
Good point! I updated the type of `UnicodeCharClass.propValue` to `Option[String]`.